src/src/com/microsoft/aad/adal/StorageHelper.java

```diff
@@ -450,6 +447,17 @@ final synchronized private SecretKey getSecretKeyFromAndroidKeyStore() throws IO
             if (encryptedKey == null || encryptedKey.length == 0) {
                 throw new UnrecoverableKeyException("Couldn't find encrypted key in file");
             }
+            
+            // Check if the retrieved keypair is empty. With the current limitation of 
+            // AndroidKeyStore, there is possibility that the alias is not wiped but 
+            // the key data is wiped, if this is the case, the retrieved keypair will
+            // be empty, and when we use the private key to do unwrap, we'll encounter 
+            // IllegalArgumentException
+            final PrivateKey privateKey = mKeyPair.getPrivate();
```

Comment:
I agree it looks like it would be cleaner if `getKeyPairFromAndroidKeyStore` would perform this verification.